test-suite/capfloor.cpp

```diff
@@ -610,6 +612,184 @@ void CapFloorTest::testCachedValue() {
             << "    expected:   " << cachedFloorNPV);
 }
 
+void CapFloorTest::testCachedValueFromOptionLets() {
+
+    BOOST_TEST_MESSAGE("Testing Black cap/floor price as a sum of optionlets prices against cached values...");
+
+    CommonVars vars;
+
+    Date cachedToday(14,March,2002),
+         cachedSettlement(18,March,2002);
+    Settings::instance().evaluationDate() = cachedToday;
+    ext::shared_ptr<YieldTermStructure> baseCurve = flatRate(cachedSettlement, 
+                                                             0.05, Actual360());                                              
+    vars.termStructure.linkTo(baseCurve);
+    Date startDate = vars.termStructure->referenceDate();
+    Leg leg = vars.makeLeg(startDate,20);  
+
+    ext::shared_ptr<Instrument> cap = vars.makeCapFloor(CapFloor::Cap,leg,
+                                                          0.07,0.20);
+    ext::shared_ptr<Instrument> floor = vars.makeCapFloor(CapFloor::Floor,leg,
+                                                            0.03,0.20);
+#ifndef QL_USE_INDEXED_COUPON
+    // par coupon price
+    Real cachedCapNPV   = 6.87570026732,
+         cachedFloorNPV = 2.65812927959;
+    Real calculatedCapletsNPV = 0.0,
```

Comment:
These two don't go into the `#ifndef`.

Reply:
`calculatedFloorletsNPV` and `calculatedCapletsNPV` moved above `#ifndef`